processing/src/main/java/io/druid/segment/DimensionHandler.java

```diff
@@ -0,0 +1,185 @@
+   *
+   * @return A new DimensionIndexer object.
+   */
+  public DimensionIndexer<EncodedType, ActualType> makeIndexer();
+
+
+  /**
+   * Creates a new DimensionMergerV9, a per-dimension object responsible for merging indexes/row data across segments
+   * and building the on-disk representation of a dimension. For use with IndexMergerV9 only.
+   *
+   * See {@link DimensionMergerV9} interface for more information.
+   *
+   * @param indexSpec     Specification object for the index merge
+   * @param outDir        Location to store files generated by the merging process
+   * @param ioPeon        ioPeon object passed in by IndexMerger, manages files created by the merging process
+   * @param capabilities  The ColumnCapabilities of the dimension represented by this DimensionHandler
+   * @param progress      ProgressIndicator used by the merging process
+
+   * @return A new DimensionMergerV9 object.
+   */
+  public DimensionMergerV9 makeMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  );
+
+
+  /**
+   * Creates a new DimensionMergerLegacy, a per-dimension object responsible for merging indexes/row data across segments
+   * and building the on-disk representation of a dimension. For use with IndexMerger only.
+   *
+   * See {@link DimensionMergerLegacy} interface for more information.
+   *
+   * @param indexSpec     Specification object for the index merge
+   * @param outDir        Location to store files generated by the merging process
+   * @param ioPeon        ioPeon object passed in by IndexMerger, manages files created by the merging process
+   * @param capabilities  The ColumnCapabilities of the dimension represented by this DimensionHandler
+   * @param progress      ProgressIndicator used by the merging process
+
+   * @return A new DimensionMergerLegacy object.
+   */
+  public DimensionMergerLegacy makeLegacyMerger(
+      IndexSpec indexSpec,
+      File outDir,
+      IOPeon ioPeon,
+      ColumnCapabilities capabilities,
+      ProgressIndicator progress
+  );
+
+  /**
+   * Returns a function that attempts to convert Object values to this dimension's type.
+   * Needed because the types of arriving row values may not always match this dimension's type during ingestion.
+   *
+   * If no transformation was performed, the transformation function must return the input object.
+   *
+   * @return A value transformation function.
+   */
+  public Function<Object, ActualType> getValueTypeTransformer();
+
+
+  /**
+   * Given an array representing a single set of row value(s) for this dimension as an Object,
+   * return the length of the array after appropriate type-casting.
+   *
+   * For example, a dictionary encoded String dimension would receive an int[] as an Object.
+   *
+   * @param dimVals Array of row values
+   * @return Size of dimVals
+   */
+  public int getLengthFromArrayObject(Object dimVals);
```

Comment:
I kept this in the Handler because it would also be used during query time (not implemented in this PR), where a DimensionIndexer would not be available.